### PR TITLE
[exporter/elasticsearchexporter]: Add dynamic index option for both logs and traces

### PR DESCRIPTION
**Pull request by leewoobin789**
**Description:** <Describe what has changed.>
The elasticsearchexporter of current version takes each one static index for both logs and traces. The capability of dynamic indexing based on attributes (<del>`logs_index_prefix`, `logs_index_suffix`, `traces_index_prefix`, `traces_index_suffix`, </del> `elasticsearch.index.prefix`, `elasticsearch.index.suffix`) will help users to delegate the final index. 
Until now, what I have seen solar is the usage of ingestion pipeline dedicatedly for rewriting index, which is the only way to write data to multiple indexes.

This way, users have multiple options to route their data to a prefixed or suffixed index by using resource processor or attribute processor in sidecar collector/ main collector or even by adding it as static attribute in application instrumentation.

**Link to tracking Issue:** <Issue number if applicable>
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/5854

**Testing:** <Describe what testing was performed and which tests were added.>
Test with test-purposed elastic search exporter with attributes defined above and check if the incoming bulk request has arrived in mock server with the expected prefixed&suffixed index.

**Documentation:** <Describe the documentation added.>
Added in Readme

### CODE REVIEW

**Comment by leewoobin789** [comment]
@MatheusGeiger that is great to hear! I am still waiting for the code review. 
@JaredTan95 FYI.

File: .chloggen/elasticsearchexporter-dynamic-index.yaml
Comment (JaredTan95) [inline .chloggen/elasticsearchexporter-dynamic-index.yaml:1]: I think this could be new `feature` 👍

**Review by JaredTan95 — COMMENTED** [review]
thx your contributions 👍
except for the above comments, other LGTM

File: .chloggen/elasticsearchexporter-dynamic-index.yaml
Comment (leewoobin789) [reply .chloggen/elasticsearchexporter-dynamic-index.yaml]: @JaredTan95 I conducted the points you mentioned. check it out :)

Comment (leewoobin789) [reply .chloggen/elasticsearchexporter-dynamic-index.yaml:1]: @JaredTan95 
```
@github-actions
build-and-test / build-package (deb) (pull_request) Failing after 2m
[Details](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/3656753677/jobs/6180031352)
@github-actions
build-and-test / build-package (rpm) (pull_request) Failing after 3m
```
looks like the pipeline or the build process is wrong.

**Review by JaredTan95 — APPROVED** [review]
LGTM

**Comment by leewoobin789** [comment]
@dmitryax this pr needs one more approval. :) thank you in advance

**Comment by leewoobin789** [comment]
just to prevent this pr being staled.

File: exporter/elasticsearchexporter/README.md
Comment (dmitryax) [reply exporter/elasticsearchexporter/README.md:29]: > however having a static variable being concatenated is good way to have a fallback in case that defined resource or attribute is not available.

Not sure I understand this concern. Wouldn't the fallback be the same value defined in `logs_index` in both cases? I'm just trying to see if there are strong reasons to provide this rather more complicated interface to users

Comment (leewoobin789) [reply exporter/elasticsearchexporter/README.md:29]: @dmitryax 
we could use instead certain resources/attributes itself as index without prefixing or suffixing `logs_index` and `traces_index` leaving `logs_index` and `traces_index` as defaults, if there are no attributes found.

however this way, there is one downside. it is harder to specify **index pattern** in elasticsearch, as there is no guarantee of having similarity in the resulting index names.

**Review by dmitryax — APPROVED** [review]

File: exporter/elasticsearchexporter/config.go
Comment (leewoobin789) [reply exporter/elasticsearchexporter/config.go:56]: @dmitryax upppps the latter one. i will correct it and test the whole thing once again locally.

Comment (leewoobin789) [reply exporter/elasticsearchexporter/config.go]: @dmitryax ✅ Done, locally tested and also bugs fixed (null pointer panic detected but fixed & tested again)

**Comment by leewoobin789** [comment]
is there anything else, which needs to be done here? :)

File: exporter/elasticsearchexporter/README.md
Comment (dmitryax) [inline exporter/elasticsearchexporter/README.md:37]: Not sure about the attributes naming:
1. `traces_`, `metric_`, `logs_` prefix seems redundant, since it's already set on spans, metrics or logs.
2. If you look at https://github.com/open-telemetry/opentelemetry-specification, attributes are usually separated by dots, not underscores.
3. Do we intend to apply the attribute to other exporters? If not, we should probably add the prefix `elasticsearch.`.

So maybe `elasticsearch.index.prefix` and `elasticsearch.index.suffix`?

Please let me know WDTY

Comment (leewoobin789) [reply exporter/elasticsearchexporter/README.md:37]: @dmitryax that makes absolutely sense to adjust the naming convention to the one defined in spec and also to consolidate the attributes to each prefix and suffix for every telemetry.
i will change it accordingly.

Comment (leewoobin789) [reply exporter/elasticsearchexporter/README.md]: @dmitryax adjusted :)

**Comment by vsellier** [comment]
Hello there,
Is there any blocker to merge this diff?

It's a important feature blocking for us to deploy opentelemetry with a production load without having a huge single index.

I volonteer to help if something more is needed :smile:

**Comment by leewoobin789** [comment]
> Hello there,
> Is there any blocker to merge this diff?
> 
> It's a important feature blocking for us to deploy opentelemetry with a production load without having a huge single index.
> 
> I volonteer to help if something more is needed :smile: 
> 

Tbh I don't think there is things to do as it is already tested. I am also waiting for approver to merge it as well.

**Comment by rpanand24** [comment]
Hi,
can someone authorized please help here to get this feature merged to main?  
Our open telemetry migration is blocked because our use case also needs dynamic indexing.

**Review by dmitryax — APPROVED** [review]

**Comment by dmitryax** [comment]
Merged. Sorry for the delay

**Pull request merged**
